paymentapi/src/test/java/fi/hel/verkkokauppa/payment/api/PaymentAdminControllerTest.java

```diff
@@ -377,6 +386,42 @@ public void whenGetPaymentMethodsThenReturnMoreThanZeroWithStatus200() {
         Assertions.assertEquals(PaymentGatewayEnum.VISMA, responsePaymentMethodDto.getGateway());
     }
 
+    @Test
+    @RunIfProfile(profile = "local")
+    public void testPaytrailSubscriptionRenewalOrderCreatedEvent() {
+        paymentAdminController.orderCreatedEventCallbackPaytrail(
+            OrderMessage
+                .builder()
+                    .eventType(EventType.SUBSCRIPTION_RENEWAL_ORDER_CREATED)
+                    .namespace("venepaikat")
+                    .orderId(UUIDGenerator.generateType4UUID().toString())
+                    .timestamp(DateTimeUtil.getDateTime())
+                    .orderType(OrderType.SUBSCRIPTION)
+                    .priceTotal("100")
+                    .priceNet("100")
+                    .priceVat("0")
+                    .cardToken(EncryptorUtil.encryptValue("2f4de4b9-94ec-4bd5-ab39-45f1f164bda0", cardTokenEncryptionPassword))
+                    .cardExpYear("2023")
+                    .cardExpMonth("11")
+                    .cardLastFourDigits("0354")
+                    .orderItemId(UUIDGenerator.generateType4UUID().toString())
+                    .vatPercentage("0")
+                    .productName("productName")
+                    .productQuantity("1")
+                    .isSubscriptionRenewalOrder(true)
+                    .subscriptionId(UUIDGenerator.generateType4UUID().toString())
+                    .userId("user")
+                    .paymentGateway(fi.hel.verkkokauppa.common.constants.PaymentGatewayEnum.PAYTRAIL)
+                    .merchantId("0_YrloUBVL9Yu6dzAXua")
```

Comment:
you could add https://github.com/City-of-Helsinki/verkkokauppa-core/blob/9efac21409a36a22e3edbf53f330ea7f035bc9f2/paymentapi/src/test/java/fi/hel/verkkokauppa/payment/testing/BaseFunctionalTest.java#L27 to fetch merchantid with namespace.